src/ctap2/commands/get_info.rs

```diff
@@ -177,62 +197,90 @@ impl<'de> Deserialize<'de> for AuthenticatorInfo {
                 let mut max_credential_id_length = None;
                 let mut transports = None;
                 let mut algorithms = None;
+                let mut max_ser_large_blob_array = None;
+                let mut force_pin_change = None;
+                let mut min_pin_length = None;
+                let mut firmware_version = None;
+                let mut max_cred_blob_length = None;
+                let mut max_rpids_for_set_min_pin_length = None;
+                let mut preferred_platform_uv_attempts = None;
+                let mut uv_modality = None;
+                let mut certifications = None;
+                let mut remaining_discoverable_credentials = None;
+                let mut vendor_prototype_config_commands = None;
                 while let Some(key) = map.next_key()? {
                     match key {
-                        1 => {
+                        0x01 => {
                             if !versions.is_empty() {
                                 return Err(serde::de::Error::duplicate_field("versions"));
                             }
                             versions = map.next_value()?;
                         }
-                        2 => {
+                        0x02 => {
                             if !extensions.is_empty() {
                                 return Err(serde::de::Error::duplicate_field("extensions"));
                             }
                             extensions = map.next_value()?;
                         }
-                        3 => {
-                            if aaguid.is_some() {
-                                return Err(serde::de::Error::duplicate_field("aaguid"));
-                            }
-                            aaguid = Some(map.next_value()?);
+                        0x03 => {
+                            parse_next_optional_value!(aaguid, map);
                         }
-                        4 => {
+                        0x04 => {
                             options = map.next_value()?;
                         }
-                        5 => {
-                            max_msg_size = Some(map.next_value()?);
+                        0x05 => {
+                            parse_next_optional_value!(max_msg_size, map);
                         }
-                        6 => {
+                        0x06 => {
+                            if !pin_protocols.is_empty() {
+                                return Err(serde::de::Error::duplicate_field("pin_protocols"));
+                            }
                             pin_protocols = map.next_value()?;
```

Comment:
The spec says `pinUvAuthProtocols` is optional. In fact it looks like `extensions`, `options`, and `maxMsgSize` are optional as well, and were in 2.0.

Reply:
`pinUvAuthProtocols` _is_ actually optional here: It is a vec, which can be empty. Same for `extensions`. `options` is using the default, unless the token sends something else. `maxMsgSize` is handled as an Option already.